blocks/editable/index.js

```diff
@@ -177,10 +178,34 @@ export default class Editable extends wp.element.Component {
 		return true;
 	}
 
+	isEndOfEditor() {
+		const range = this.editor.selection.getRng();
+		if ( range.endOffset !== range.endContainer.textContent.length || ! range.collapsed ) {
+			return false;
+		}
+		const start = range.endContainer;
+		const body = this.editor.getBody();
+		let element = start;
+		while ( element !== body ) {
+			const child = element;
+			element = element.parentNode;
+			if ( element.lastChild !== child ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
 	onKeyDown( event ) {
-		if ( this.props.onMerge && event.keyCode === KEYCODE_BACKSPACE && this.isStartOfEditor() ) {
+		if (
+			this.props.onMerge && (
+				( event.keyCode === KEYCODE_BACKSPACE && this.isStartOfEditor() ) ||
+				( event.keyCode === KEYCODE_DELETE && this.isEndOfEditor() )
+			)
+		) {
+			const forward = event.keyCode === KEYCODE_DELETE;
 			this.onChange();
-			this.props.onMerge( this.editor.getContent() );
+			this.props.onMerge( forward );
```

Comment:
Yes, but it's not that simple. Imagine a block having multiple editables. So the merge of the last one would probably merge its content with the previous one (or do nothing) but the merge of the first editable would actually trigger the merge with the previous block. 

It could be way simpler if it's one block = one editable, but that's not the case. Could we provide more "metadata" on how the block uses the editable in this case and fallback to one editable mergeable with the previous block if this metadata is not provided? I don't know but that's something we should think about.

example: 

```js
edit( { mergeBlocks } ) {
  return (
    <div>
       <Editable value={ attributes.content } mergeWithPrevious />
       <Editable onMerge={ mergeWithAttribute( 'content' ) } />
       <Editable noMerge />
    </div>
  );
}
```